DNS Name is not canonical error message

I am trying to run acceptance tests. I've prepared an empty PowerDNS (4.0.5) instance and its API is available on `http://127.0.0.1:8081`. Now when I run tests, i get:
```
$ make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v  -timeout 120m
?   	github.com/terraform-providers/terraform-provider-powerdns	[no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProviderImpl
--- PASS: TestProviderImpl (0.00s)
=== RUN   TestAccPDNSRecord_A
--- FAIL: TestAccPDNSRecord_A (0.02s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-a: 1 error(s) occurred:

		* powerdns_record.test-a: Failed to create PowerDNS Record: Error creating record set: redis.sysa.xyz:::A, reason: "DNS Name 'redis.sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_WithPtr
--- FAIL: TestAccPDNSRecord_WithPtr (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-a-ptr: 1 error(s) occurred:

		* powerdns_record.test-a-ptr: Failed to create PowerDNS Record: Error creating record set: redis.sysa.xyz:::A, reason: "DNS Name 'redis.sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_WithCount
--- FAIL: TestAccPDNSRecord_WithCount (0.01s)
	testing.go:434: Step 0 error: Error applying: 2 error(s) occurred:

		* powerdns_record.test-counted[0]: 1 error(s) occurred:

		* powerdns_record.test-counted.0: Failed to create PowerDNS Record: Error creating record set: redis-0.sysa.xyz:::A, reason: "DNS Name 'redis-0.sysa.xyz' is not canonical"
		* powerdns_record.test-counted[1]: 1 error(s) occurred:

		* powerdns_record.test-counted.1: Failed to create PowerDNS Record: Error creating record set: redis-1.sysa.xyz:::A, reason: "DNS Name 'redis-1.sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_AAAA
--- FAIL: TestAccPDNSRecord_AAAA (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-aaaa: 1 error(s) occurred:

		* powerdns_record.test-aaaa: Failed to create PowerDNS Record: Error creating record set: redis.sysa.xyz:::AAAA, reason: "DNS Name 'redis.sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_CNAME
--- FAIL: TestAccPDNSRecord_CNAME (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-cname: 1 error(s) occurred:

		* powerdns_record.test-cname: Failed to create PowerDNS Record: Error creating record set: redis.sysa.xyz:::CNAME, reason: "DNS Name 'redis.sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_HINFO
--- FAIL: TestAccPDNSRecord_HINFO (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-hinfo: 1 error(s) occurred:

		* powerdns_record.test-hinfo: Failed to create PowerDNS Record: Error creating record set: redis.sysa.xyz:::HINFO, reason: "DNS Name 'redis.sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_LOC
--- FAIL: TestAccPDNSRecord_LOC (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-loc: 1 error(s) occurred:

		* powerdns_record.test-loc: Failed to create PowerDNS Record: Error creating record set: redis.sysa.xyz:::LOC, reason: "DNS Name 'redis.sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_MX
--- FAIL: TestAccPDNSRecord_MX (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-mx: 1 error(s) occurred:

		* powerdns_record.test-mx: Failed to create PowerDNS Record: Error creating record set: sysa.xyz:::MX, reason: "DNS Name 'sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_NAPTR
--- FAIL: TestAccPDNSRecord_NAPTR (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-naptr: 1 error(s) occurred:

		* powerdns_record.test-naptr: Failed to create PowerDNS Record: Error creating record set: sysa.xyz:::NAPTR, reason: "DNS Name 'sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_NS
--- FAIL: TestAccPDNSRecord_NS (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-ns: 1 error(s) occurred:

		* powerdns_record.test-ns: Failed to create PowerDNS Record: Error creating record set: lab.sysa.xyz:::NS, reason: "DNS Name 'lab.sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_SPF
--- FAIL: TestAccPDNSRecord_SPF (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-spf: 1 error(s) occurred:

		* powerdns_record.test-spf: Failed to create PowerDNS Record: Error creating record set: sysa.xyz:::SPF, reason: "DNS Name 'sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_SSHFP
--- FAIL: TestAccPDNSRecord_SSHFP (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-sshfp: 1 error(s) occurred:

		* powerdns_record.test-sshfp: Failed to create PowerDNS Record: Error creating record set: ssh.sysa.xyz:::SSHFP, reason: "DNS Name 'ssh.sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_SRV
--- FAIL: TestAccPDNSRecord_SRV (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-srv: 1 error(s) occurred:

		* powerdns_record.test-srv: Failed to create PowerDNS Record: Error creating record set: _redis._tcp.sysa.xyz:::SRV, reason: "DNS Name '_redis._tcp.sysa.xyz' is not canonical"
=== RUN   TestAccPDNSRecord_TXT
--- FAIL: TestAccPDNSRecord_TXT (0.01s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* powerdns_record.test-txt: 1 error(s) occurred:

		* powerdns_record.test-txt: Failed to create PowerDNS Record: Error creating record set: text.sysa.xyz:::TXT, reason: "DNS Name 'text.sysa.xyz' is not canonical"
FAIL
FAIL	github.com/terraform-providers/terraform-provider-powerdns/powerdns	0.177s
make: *** [testacc] Error 1
```

### Affected Resource(s)
- powerdns_record

### Terraform Configuration Files
```hcl
resource "powerdns_record" "test-a" {
  zone = "sysa.xyz"
	name = "redis.sysa.xyz"
	type = "A"
	ttl = 60
	records = [ "1.1.1.1", "2.2.2.2" ]
}
```

### Expected Behavior, Actual Behavior
Acceptance tests should have passed. Adding a dot at the end of `name` field fixes the problem.

So, my question is:
- If this is expected, should we fix hcl strings in `resource_powerdns_record.go` to have dots in name?
- If not expected, maybe add a dot substitution inside a code?

I would gladly make a PR, but let's discuss, first.

--- a/powerdns/resource_powerdns_record_test.go
+++ b/powerdns/resource_powerdns_record_test.go
@@ -272,8 +272,8 @@ func testAccCheckPDNSRecordExists(n string) resource.TestCheckFunc {
 
 const testPDNSRecordConfigA = `
 resource "powerdns_record" "test-a" {
-  zone = "sysa.xyz"
-	name = "redis.sysa.xyz"
+	zone = "sysa.xyz."
+	name = "redis.sysa.xyz."
 	type = "A"
 	ttl = 60
 	records = [ "1.1.1.1", "2.2.2.2" ]
@@ -282,8 +282,8 @@ resource "powerdns_record" "test-a" {
 const testPDNSRecordConfigHyphenedWithCount = `
 resource "powerdns_record" "test-counted" {
 	count = "2"
-	zone = "sysa.xyz"
-	name = "redis-${count.index}.sysa.xyz"
+	zone = "sysa.xyz."
+	name = "redis-${count.index}.sysa.xyz."
 	type = "A"
 	ttl = 60
 	records = [ "1.1.1.${count.index}" ]
@@ -291,26 +291,26 @@ resource "powerdns_record" "test-counted" {
 
 const testPDNSRecordConfigAAAA = `
 resource "powerdns_record" "test-aaaa" {
-  zone = "sysa.xyz"
-	name = "redis.sysa.xyz"
+	zone = "sysa.xyz."
+	name = "redis.sysa.xyz."
 	type = "AAAA"
 	ttl = 60
-	records = [ "2001:DB8:2000:bf0::1", "2001:DB8:2000:bf1::1" ]
+	records = [ "2001:db8:2000:bf0::1", "2001:db8:2000:bf1::1" ]
 }`
 
 const testPDNSRecordConfigCNAME = `
 resource "powerdns_record" "test-cname" {
-  zone = "sysa.xyz"
-	name = "redis.sysa.xyz"
+	zone = "sysa.xyz."
+	name = "redis.sysa.xyz."
 	type = "CNAME"
 	ttl = 60
-	records = [ "redis.example.com" ]
+	records = [ "redis.example.com." ]
 }`
 
 const testPDNSRecordConfigHINFO = `
 resource "powerdns_record" "test-hinfo" {
-  zone = "sysa.xyz"
-	name = "redis.sysa.xyz"
+	zone = "sysa.xyz."
+	name = "redis.sysa.xyz."
 	type = "HINFO"
 	ttl = 60
 	records = [ "\"PC-Intel-2.4ghz\" \"Linux\"" ]
@@ -318,8 +318,8 @@ resource "powerdns_record" "test-hinfo" {
 
 const testPDNSRecordConfigLOC = `
 resource "powerdns_record" "test-loc" {
-  zone = "sysa.xyz"
-	name = "redis.sysa.xyz"
+	zone = "sysa.xyz."
+	name = "redis.sysa.xyz."
 	type = "LOC"
 	ttl = 60
 	records = [ "51 56 0.123 N 5 54 0.000 E 4.00m 1.00m 10000.00m 10.00m" ]
@@ -327,26 +327,26 @@ resource "powerdns_record" "test-loc" {
 
 const testPDNSRecordConfigMX = `
 resource "powerdns_record" "test-mx" {
-  zone = "sysa.xyz"
-	name = "sysa.xyz"
+	zone = "sysa.xyz."
+	name = "sysa.xyz."
 	type = "MX"
 	ttl = 60
-	records = [ "10 mail.example.com" ]
+	records = [ "10 mail.example.com." ]
 }`
 
 const testPDNSRecordConfigMXMulti = `
 resource "powerdns_record" "test-mx-multi" {
-  zone = "sysa.xyz"
-	name = "sysa.xyz"
+	zone = "sysa.xyz."
+	name = "sysa.xyz."
 	type = "MX"
 	ttl = 60
-	records = [ "10 mail1.example.com", "20 mail2.example.com" ]
+	records = [ "10 mail1.example.com.", "20 mail2.example.com." ]
 }`
 
 const testPDNSRecordConfigNAPTR = `
 resource "powerdns_record" "test-naptr" {
-  zone = "sysa.xyz"
-	name = "sysa.xyz"
+	zone = "sysa.xyz"
+	name = "sysa.xyz."
 	type = "NAPTR"
 	ttl = 60
 	records = [ "100 50 \"s\" \"z3950+I2L+I2C\" \"\" _z3950._tcp.gatech.edu'." ]
@@ -354,17 +354,17 @@ resource "powerdns_record" "test-naptr" {
 
 const testPDNSRecordConfigNS = `
 resource "powerdns_record" "test-ns" {
-  zone = "sysa.xyz"
-	name = "lab.sysa.xyz"
+	zone = "sysa.xyz."
+	name = "lab.sysa.xyz."
 	type = "NS"
 	ttl = 60
-	records = [ "ns1.sysa.xyz", "ns2.sysa.xyz" ]
+	records = [ "ns1.sysa.xyz.", "ns2.sysa.xyz." ]
 }`
 
 const testPDNSRecordConfigSPF = `
 resource "powerdns_record" "test-spf" {
-  zone = "sysa.xyz"
-	name = "sysa.xyz"
+	zone = "sysa.xyz."
+	name = "sysa.xyz."
 	type = "SPF"
 	ttl = 60
 	records = [ "\"v=spf1 +all\"" ]
@@ -372,8 +372,8 @@ resource "powerdns_record" "test-spf" {
 
 const testPDNSRecordConfigSSHFP = `
 resource "powerdns_record" "test-sshfp" {
-  zone = "sysa.xyz"
-	name = "ssh.sysa.xyz"
+	zone = "sysa.xyz."
+	name = "ssh.sysa.xyz."
 	type = "SSHFP"
 	ttl = 60
 	records = [ "1 1 123456789abcdef67890123456789abcdef67890" ]
@@ -381,17 +381,17 @@ resource "powerdns_record" "test-sshfp" {
 
 const testPDNSRecordConfigSRV = `
 resource "powerdns_record" "test-srv" {
-  zone = "sysa.xyz"
-	name = "_redis._tcp.sysa.xyz"
+	zone = "sysa.xyz."
+	name = "_redis._tcp.sysa.xyz."
 	type = "SRV"
 	ttl = 60
-	records = [ "0 10 6379 redis1.sysa.xyz", "0 10 6379 redis2.sysa.xyz", "10 10 6379 redis-replica.sysa.xyz" ]
+	records = [ "0 10 6379 redis1.sysa.xyz.", "0 10 6379 redis2.sysa.xyz.", "10 10 6379 redis-replica.sysa.xyz." ]
 }`
 
 const testPDNSRecordConfigTXT = `
 resource "powerdns_record" "test-txt" {
-  zone = "sysa.xyz"
-	name = "text.sysa.xyz"
+	zone = "sysa.xyz."
+	name = "text.sysa.xyz."
 	type = "TXT"
 	ttl = 60
 	records = [ "\"text record payload\"" ]

--- a/website/docs/r/record.html.markdown
+++ b/website/docs/r/record.html.markdown
@@ -20,7 +20,7 @@ For the v1 API (PowerDNS version 4):
 # Add a record to the zone
 resource "powerdns_record" "foobar" {
   zone    = "example.com."
-  name    = "www.example.com"
+  name    = "www.example.com."
   type    = "A"
   ttl     = 300
   records = ["192.168.0.11"]
@@ -44,8 +44,8 @@ For the legacy API (PowerDNS version 3.4):
 ```hcl
 # Add a record to the zone
 resource "powerdns_record" "foobar" {
-  zone    = "example.com"
-  name    = "www.example.com"
+  zone    = "example.com."
+  name    = "www.example.com."
   type    = "A"
   ttl     = 300
   records = ["192.168.0.11"]
@@ -61,4 +61,3 @@ The following arguments are supported:
 * `type` - (Required) The record type.
 * `ttl` - (Required) The TTL of the record.
 * `records` - (Required) A string list of records.
-

